bcdaworker/worker/worker.go

```diff
@@ -164,20 +166,41 @@ func (w *worker) ProcessJob(ctx context.Context, job models.Job, jobArgs models.
 	return nil
 }
 
-func moveFiles(tempDir string, stagingDir string) error {
+func compressFiles(tempDir string, stagingDir string) error {
 	// Open the input file
 	files, err := os.ReadDir(tempDir)
 	if err != nil {
 		err = errors.Wrap(err, "Error reading from the staging directory for files for Job")
 		return err
 	}
+	gzipLevel, err := strconv.Atoi(os.Getenv("COMPRESSION_LEVEL"))
+	if err != nil || gzipLevel < 1 || gzipLevel > 9 { //levels 1-9 supported by BCDA.
+		gzipLevel = gzip.DefaultCompression
+	}
 	for _, f := range files {
 		oldPath := fmt.Sprintf("%s/%s", tempDir, f.Name())
 		newPath := fmt.Sprintf("%s/%s", stagingDir, f.Name())
-		err := os.Rename(oldPath, newPath) //#nosec G304
+		inputFile, err := os.Open(oldPath) //#nosec G304
+		if err != nil {
+			return err
+		}
+		defer inputFile.Close() //#nosec G307
```

Comment:
I know this is essentially safe but would prefer to avoid `#nosec` markers if possible from an auditing perspective. Wonder if we want to create a helper utility for these scenarios?

```golang
// usage: defer CloseOrLogError(inputFile)
func CloseOrLogError(f *os.File) {
	if f == nil {
		return
	}
	if err := f.Close(); err != nil {
		logger.Warnf("Error closing file: %v", err)
	}
}
```

Reply:
Good idea; I modified this function slightly, to utilize the snippet above (and passing a field logger to enable the logging).